narwhals/dataframe.py

```diff
@@ -157,6 +157,150 @@ def join(
 
 
 class DataFrame(BaseFrame):
+    r"""
+    Two-dimensional data structure representing data as a table with rows and columns.
+
+    Arguments:
+        df: A pandas-like dataframe (Pandas, cuDF or Modin), a Polars dataframe,
+             a narwhals DataFrame or a narwhals LazyFrame.
+
+        is_polars: if set to `True`, assume the dataframe to be of Polars type.
+
+    Examples:
+        Constructing a DataFrame from a dictionary:
+
+        >>> import polars as pl
+        >>> import narwhals as nw
+        >>> data = {"a": [1, 2], "b": [3, 4]}
+        >>> df_pl = pl.DataFrame(data)
+        >>> df = nw.DataFrame(df_pl)
+        >>> df
+        ┌─────────────────────────────────────────────────┐
+        | Narwhals DataFrame                              |
+        | Use `narwhals.to_native()` to see native output |
+        └─────────────────────────────────────────────────┘
+        >>> nw.to_native(df)
+        shape: (2, 2)
+        ┌─────┬─────┐
+        │ a   ┆ b   │
+        │ --- ┆ --- │
+        │ i64 ┆ i64 │
+        ╞═════╪═════╡
+        │ 1   ┆ 3   │
+        │ 2   ┆ 4   │
+        └─────┴─────┘
+
+        To specify a more detailed/specific frame schema you can supply the `schema`
+        parameter with a dictionary of (name,dtype) pairs...
+
+        >>> data = {"col1": [0, 2], "col2": [3, 7]}
+        >>> df_pl2 = pl.DataFrame(data, schema={"col1": pl.Float32, "col2": pl.Int64})
```

Comment:
this only works for Polars though, right? I don't think we should document it if it's Polars-specific

Same with the rest of the examples - they're showing how to create a Polars dataframe, which is then just passed to Narwhals. I think we can skip them, the initial simple example is enough

Reply:
Yes, makes sense